hathor/transaction/transaction_metadata.py

```diff
@@ -229,9 +209,12 @@ def to_json(self) -> dict[str, Any]:
         data['twins'] = [x.hex() for x in self.twins]
         data['accumulated_weight'] = self.accumulated_weight
         data['score'] = self.score
-        data['height'] = self.height
         data['min_height'] = self.min_height
-        data['feature_activation_bit_counts'] = self.feature_activation_bit_counts
+
+        from hathor.transaction import Block
+        if isinstance(self.get_tx(), Block):
+            data['height'] = self.height
```

Comment:
This might be a breaking chance for the APIs. I guess it needs further analysis.

Reply:
Yes, but clients shouldn't be using heights for transactions as they were always `0`. Indeed it may break if some client uses some kind of schema validation and expects that `height` is always set. We can either leave it like this and validate it during QA, or I can change it to keep sending `0` for compatibility (just like I did for events as explained in https://github.com/HathorNetwork/hathor-core/pull/1014#discussion_r1715790521).

Reply:
I guess we should add a list of incompatibilities in the acceptance criteria for discussion. My initial guess is that we should list all breaking changes of a new release. We might even be better bumping the major version when we do it. Not sure if it's worth doing it because of this refactor, though.

Reply:
I added a "Breaking Changes" section to the PR description.

Reply:
I would rather keep `"height": 0` in the API and add a comment for we to remove it in the future. Since it looks like we have no data about whether this can break any schema validation and keeping it doesn't seem to have any negative impact.

Similarly in the #572 I would like to remove `score`/`accumulated_weight` in favor of `score_raw`/`accumulated_weight_raw` in the future, but all these changes could be introduced at once, possibly with a toggle.

Reply:
Sure, done in https://github.com/HathorNetwork/hathor-core/pull/1014/commits/6866b723f902b7960e46df61bb33fd08c0ab1839

Reply:
Thanks!